File: src/main/services/config-service.ts
import { join } from "node:path";
import { app } from "electron";
import { Config } from "../models/config";
import { ConfigRepository } from "../repositories/config-repository";

export class ConfigService {
  constructor(private readonly repository: ConfigRepository) {}

  loadConfig(): Config {
    let config = this.repository.load();
    if (!config) {
      config = new Config();
    }
    if (!config.mod_folder) {
      config.mod_folder = join(app.getPath("userData"), "mods");
    }
    config.normalizePaths();
    return config;
  }

  saveConfig(config: Config): void {
    config.normalizePaths();
    this.repository.save(config);
  }

  validateConfig(config: Config): boolean {
    return config.isValid();
  }
}

File: .mewtator/app/core/services/config_service.py
from app.core.models.config import Config
from app.infrastructure.config_repository import ConfigRepository


class ConfigService:
    def __init__(self, repository: ConfigRepository):
        self.repository = repository
    
    def load_config(self) -> Config:
        config = self.repository.load()
        
        if config is None:
            config = Config()
        
        if not config.mod_folder:
            import os
            config.mod_folder = os.path.join(os.getcwd(), "mods")
        
        config.normalize_paths()
        return config
    
    def save_config(self, config: Config):
        config.normalize_paths()
        self.repository.save(config)
    
    def validate_config(self, config: Config) -> bool:
        return config.is_valid()
